cosa/20200804-cfg-spec.md

```diff
@@ -0,0 +1,107 @@
+# CoreOS Assemlber Config Specification
+- Subject ID: COSA.20200804
+- Superceeds: None
+- Conflicts: None
+
+## Summary
+
+The RHCOS development pipelines use a "jobspec" to describe the pipeline runs.  This enhancement proposes a file-based interface based loosely on the RHCOS jobspec.
+
+## Goals
+
+1. Provide a file-based interface into COSA
+1. Prep for breaking up the Pipeline monoliths
+1. Provide a migration path for RHCOS pipelines
+1. Prepare for pipeline Nirvana (where *COS pipelines can live in peace and harmony)
+
+## Why?
+
+The innovation of the RHCOS JobSpec was it provided a clear seperation of configuration from the code; the logic was conflated. Further the jobspec allowed the logic to be run with different inputs.
+
+To support this, a "COSAspec" is being introduced to provide a CLI file-interface.
```

Comment:
Can you include a config name for the YAML file so that we can bikeshed on it? :art:
I'll throw in a suggestion for `cosa-config.yaml`.

Also, this proposal doesn't cover *where* this config needs to live. Is it auto-discovered by `cosa`? If so, where does it look?

Reply:
I intentionally left that the where and name of the config out. The rationale is that the COSAspec YAML is something that is brought to COSA. If anything I would run with the idea of it coming from an external repo (e.g. f.c.c or rh.c.c).